### PR TITLE
Add packages support for security policy

### PR DESCRIPTION
**Pull request by dortam888**
Added package_name, package_type, and package_versions which are available from Xray 3.87.5 to security policy criteria.

### CODE REVIEW

File: pkg/xray/resource/resource_xray_custom_issue.go [hunk 1]
```diff
@@ -17,37 +17,28 @@ import (
 
 var validPackageTypes = []string{
 	"alpine",
-	"bower",
+	//"bower",
```
Comment (alexhung) [inline pkg/xray/resource/resource_xray_custom_issue.go:20]: Let's delete these commented out lines.

Comment (alexhung) [reply pkg/xray/resource/resource_xray_custom_issue.go:20]: We need to generate updated documentation. You can do this by:

```shell
make doc
```

Comment (alexhung) [reply pkg/xray/resource/resource_xray_custom_issue.go:20]: Is this list the same as the one for security policy? We may need 2 separate lists if they aren't the same.

File: pkg/xray/resource/resource_xray_security_policy_test.go [hunk 2]
```diff
@@ -1168,3 +1240,30 @@ const securityPolicyMaliciousPkgFixVersionDep = `resource "xray_security_policy"
...
+			notify_deployer = {{ .notify_deployer }}
+			create_ticket_enabled = {{ .create_ticket_enabled }}
+			build_failure_grace_period_in_days = {{ .grace_period_days }}
+			block_download {
+				unscanned = {{ .block_unscanned }}
+				active = {{ .block_active }}
+			}
+		}
+	}
+}`
```
Comment (alexhung) [inline pkg/xray/resource/resource_xray_security_policy_test.go:1269]: I think the tests for invalid attributes combos are missing, i.e. logic from the custom diff func.

**Review by alexhung — CHANGES_REQUESTED** [review]
Good start. Just a few minor things to fix, as well as updating the doc.

File: pkg/xray/resource/resource_xray_custom_issue.go (hunk 1 above)
Comment (dortam888) [reply pkg/xray/resource/resource_xray_custom_issue.go:20]: > Is this list the same as the one for security policy? We may need 2 separate lists if they aren't the same.

Lists should be the same according to product. 
Currently, the commented-out package types are for some reason not supported in v2/policies API (Product could not give me an answer as to why). We can include them for future support or remove them and create separate lists for both APIs. Also might consider moving validPackageTypes to another file if both APIs are using it.

Comment (alexhung) [reply pkg/xray/resource/resource_xray_custom_issue.go:20]: I think having 2 separate lists is best for now until we know for sure they are same list.

**Review by alexhung — APPROVED** [review]
LGTM

File: docs/resources/security_policy.md [hunk 3]
```diff
@@ -188,6 +188,9 @@ Optional:
 - `fix_version_dependant` (Boolean) Default value is `false`. Issues that do not have a fixed version are not generated until a fixed version is available. Must be `false` with `malicious_package` enabled.
 - `malicious_package` (Boolean) Default value is `false`. Generating a violation on a malicious package.
 - `min_severity` (String) The minimum security vulnerability severity that will be impacted by the policy. Valid values: `All Severities`, `Critical`, `High`, `Medium`, `Low`
+- `package_name` (String) The package name to create a rule for
+- `package_type` (String) The package type to create a rule for
+- `package_versions` (Set of String) package versions to apply the rule on can be (,) for any version or a open range (1,4) or closed [1,4] or one version [1]
```
Comment (danielmkn) [inline docs/resources/security_policy.md:193]: a open range -> an open range

**Review by danielmkn — CHANGES_REQUESTED** [review]
Very small one, please fix the typo.

File: pkg/xray/resource/resource_xray_security_policy.go [hunk 4]
```diff
@@ -106,6 +106,28 @@ func ResourceXraySecurityPolicyV2() *schema.Resource {
...
+			ValidateDiagFunc: validator.StringInSlice(true, validPackageTypesSupportedXraySecPolicies...),
+		},
+		"package_versions": {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "package versions to apply the rule on can be (,) for any version or an open range (1,4) or closed [1,4] or one version [1]",
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+				ValidateDiagFunc: validation.ToDiagFunc(
+					validation.StringMatch(regexp.MustCompile(`((^(\(|\[)((\d+\.)?(\d+\.)?(\*|\d+)|(\s*))\,((\d+\.)?(\d+\.)?(\*|\d+)|(\s*))(\)|\])$|^\[(\d+\.)?(\d+\.)?(\*|\d+)\]$))`), "invalid Range, must be one of the follows: Any Version: (,) or Speciific Version: [1.2], [3] or Range: (1,), [,1.2.3], (4.5.0,6.5.2]"),
```
Comment (alexhung) [inline pkg/xray/resource/resource_xray_security_policy.go:127]: Sorry, just spotted `Speciific` typo 😄

**Review by alexhung — CHANGES_REQUESTED** [review]
One more typo!

**Review by alexhung — APPROVED** [review]

**Pull request merged**